// simple-joy-api/dist/models/Post/Content/ContentElement.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class ContentElement {
    constructor() {
        this._type = null;
        this._data = null;
        this._tag = null;
        this._link = null;
        this._childrens = null;
    }
    get childrens() {
        return this._childrens;
    }
    set childrens(value) {
        this._childrens = value;
    }
    get tag() {
        return this._tag;
    }
    set tag(value) {
        this._tag = value;
    }
    get type() {
        return this._type;
    }
    set type(value) {
        this._type = value;
    }
    get data() {
        return this._data;
    }
    set data(value) {
        this._data = value;
    }
    get link() {
        return this._link;
    }
    set link(value) {
        this._link = value;
    }
}
exports.default = ContentElement;
ContentElement.IMAGE_TYPE = 'image';
ContentElement.IFRAME_TYPE = 'iframe';
ContentElement.VIDEO_TYPE = 'video';
ContentElement.OTHER_TYPE = 'other';
ContentElement.LINK_TYPE = 'link';
