// simple-joy-api/dist/models/Post.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
class Post {
    constructor() {
        this._id = null;
        this._content = null;
        this._author = null;
        this._tags = [];
        this._date = null;
        this._commentsCount = null;
        this._rating = null;
    }
    get id() {
        return this._id;
    }
    set id(value) {
        this._id = value;
    }
    get content() {
        return this._content;
    }
    set content(value) {
        this._content = value;
    }
    get author() {
        return this._author;
    }
    set author(value) {
        this._author = value;
    }
    get tags() {
        return this._tags;
    }
    set tags(value) {
        this._tags = value;
    }
    get date() {
        return this._date;
    }
    set date(value) {
        this._date = value;
    }
    get commentsCount() {
        return this._commentsCount;
    }
    set commentsCount(value) {
        this._commentsCount = value;
    }
    get rating() {
        return this._rating;
    }
    set rating(value) {
        this._rating = value;
    }
}
exports.default = Post;
